Fix support login configuration

steventux opened the pull request:
### Context

We configure a separate set of authentication routes and omniauth properties for support users.
This complexity isn't strictly necessary as we can simply examine the authenticated user's roles for an internal user role and deem them to be support users.

<!-- Why are you making this change? -->

### Changes proposed in this pull request

Refactor Omniauth config and associated code to use the same authentication callback.

<!-- Include a summary of the change. -->
<!-- Why this particular solution? -->
<!-- What assumptions have you made? -->
<!-- Are there any side effects to note? -->
<!-- If there are UI changes, please include Before and After screenshots. -->

### Guidance to review

This _should just work_ if you set your local `DSI_BYPASS=false`. Dev DSI config remains unchanged. 
You need to sign in via the `/check-records/sign-in` path and the policies/org/roles should take care of whether you are given access to the support UI.

<!-- How could someone else check this work? -->
<!-- Which parts do you want more feedback on? -->

### Link to Trello card

https://trello.com/c/3l2MV3KD/1758-fix-support-login-configuration
<!-- http://trello.com/123-example-card -->

### Checklist

- [x] Attach to Trello card
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally

steventux committed:
Simplify omniauth config

Remove the secondary staff omniauth config.

steventux committed:
Remove staff omniauth callback controller

We'll use the check callback controller for all authentication.

steventux committed:
Remove staff sign in and not authorised templates

steventux committed:
Remove staff access check service class

steventux committed:
Remove staff column from dsi_users

steventux committed:
Remove support sign in and sign out routes and actions

steventux committed:
Remove DsiUser.staff scope and dependents

steventux committed:
Amend navigation to work without support namespace

We can't rely on the support namespace to denote a support user as they now use the check records signin actions.
Use current_staff as the way to determine which nav items to show.

steventux committed:
Redirect internal users to the appropriate root path

On successful authentication examine the user for the internal role code and redirect accordingly.

steventux committed:
Amend internal role code env var

This makes it consistent with other services.

steventux committed:
Ensure we check for the internal role code before any others

This makes sure the user is granted the maximum app permissions if they have the internal role code.

steventux committed:
Refactor specs to work with simplified support authentication

steventux committed:
Amend Staff controller to use DsiUser.internal scope

gpeng commented on app/models/dsi_user.rb:
```diff
@@ -5,16 +5,19 @@ class DsiUser < ApplicationRecord
   has_many :dsi_user_sessions, dependent: :destroy
 
-  scope :staff, -> { where(staff: true) }
+  scope :internal, -> {
```
Would this end up returning users as internal if they'd previously logged in as an internal user? I don't suppose it matters too much given we're only using it to list them in the support console. Maybe there's no need for that page anymore given we can no longer add or remove them inside the service. Is that page just left over from when we used devise for the staff users?

steventux replied:
Yes the page is an artefact of devise and the various invitations and password reset functions, we could just remove all that and do away with the scope.

steventux replied:
Removed Staff index here: https://github.com/DFE-Digital/access-your-teaching-qualifications/pull/550/commits/945640508b220bba6875ed4bae48976dbb6183c7

gpeng replied:
I don't think this scope is used any more then is it? Can we remove it altogether?

gpeng commented on app/controllers/support_interface/support_interface_controller.rb:
```diff
@@ -13,7 +13,10 @@ class SupportInterfaceController < ApplicationController
     layout "support_layout"
 
     def current_staff
```
Do we need this anymore? Should the support interface just have it's own authorise code that checks `current_dsi_user.internal?`

steventux replied:
@gpeng It's used by `find_current_auditor` which is a `audits1984` requirement according to https://github.com/DFE-Digital/access-your-teaching-qualifications/commit/d98fd0c3fe1046cc009be7d3456d3c4a489a4366 and also to determine which navigation items to show. We could rename it do be less Devise-y?

steventux replied:
Removed `current_staff` here https://github.com/DFE-Digital/access-your-teaching-qualifications/pull/550/commits/da2aa4ea598e2dcdf50022a4164a636b3dcd0238

steventux committed:
Remove current_staff method

This isn't necessary and only serves as a wrapper around DsiUser#internal?

steventux committed:
Remove support interface staff index

This was an artefact of using Devise to manage Staff password resets and invitations, we don't need it any more.

gpeng reviewed (approved):
Would be good to get @malcolmbaig to have a quick look too and test on non-prod before we merge given this is quite a sprawling change.
